### ISSUE TITLE
Minimalist binder image

### ISSUE THREAD

**Issue by jlperla** [QuantEcon/lecture-source-jl#657]
I looked at a few of the docker stacks images, and they are pretty massive.  Here is my analysis:
- 227MB: https://hub.docker.com/r/jupyter/base-notebook/tags/  is .  Doesn't look like any garbage there as it just sets up miniconda+jupyter stuff.  Good place to start
- 1GB: https://hub.docker.com/r/jupyter/minimal-notebook/dockerfile adds a bunch to base, but the biggest stuff there may be the latex/pandoc.  I don't think that is necessary unless you are saving out latex files.  My suggestion is just to take a subset such as
```
RUN apt-get update && apt-get install -yq --no-install-recommends \
    build-essential \
    git \
    libsm6 \
    libxext-dev \
    libxrender1 \
    netcat \
    tzdata \
    unzip \
    nano \
    && rm -rf /var/lib/apt/lists/*
```
Or something like that.  I bet it is WAY smaller.
- 1GB? https://hub.docker.com/r/jupyter/scipy-notebook/dockerfile  adds stuff to minimal.  But I don't think we need any of that for a julia build.  I would suggest doing something like
```
RUN jupyter labextension install @jupyter-widgets/jupyterlab-manager@^1.0.1 --no-build && \
    jupyter lab build --dev-build=False && \
    npm cache clean --force && \
    rm -rf $CONDA_DIR/share/jupyter/lab/staging && \
    rm -rf /home/$NB_USER/.cache/yarn && \
    rm -rf /home/$NB_USER/.node-gyp && \
    fix-permissions $CONDA_DIR && \
    fix-permissions /home/$NB_USER
```
I don't think we need anything else?
- 2GB Datascience:  adds to the scipy with a bunch of stuff, including fortran and R.  See https://hub.docker.com/r/jupyter/datascience-notebook/dockerfile
  - I am not sure what of this we really need for a tightly controlled lecture set.  The majority of this stuff may be `r` stuff...  If we strip down the julia it may be more manageable.
  - Otherwise, I think all of the existing julia stuff is relevant, but I am not sure what else we need?

The image won't be tiny purely due to the Julia precompilation, but it may not be that bad.

And that is a final thing to try.  Instead of precompiling everything, just precompile the packages we think we really need easily accessible, and leave out everything else that may create massive files.

Basically, we should do a `]activate .; instantiate` the core `.toml` file, but not precompile.  Then instead of calling precomplile, do a `using` with the activated project for key packages.  For example, I think this covers the vast majority of notebooks outside of the `more_julia` section (which is a reasonable target).
```julia
using Plots, Distributions, NLsolve, StaticArrays, ForwardDiff, QuantEcon, Parameters, Expectations, StatsPlots, Roots, Interpolations, Optim
```
We can see if that helps significantly.  I would rather have a few of the notebooks having precompile when the users run them than add a gig to the image.

**Comment by arnavs** [comment]
I'm closing this since it is implemented, and works well per my testing. 

Once @jlperla  experiments, he can make a decision about whether this is preferable to the standard image.
